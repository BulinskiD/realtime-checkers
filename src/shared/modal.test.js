import React from 'react';
import InfoModal from './modal';
import {shallow} from "enzyme";


describe('InfoModal', ()=>{

    it('should render correctly with given props', ()=>{
        const clickFn = jest.fn();
        const component = shallow(<InfoModal onClose={clickFn} title="Test" buttonText="Button">Error</InfoModal>);
        expect(component.find("ModalBody").text()).toBe("Error");
        expect(component.find("ModalTitle").text()).toBe("Test");
        expect(component.find("Button").text()).toBe("Button");
        expect(component).toMatchSnapshot();
    });

    it('should invoke clickFn when button is clicked', ()=>{
        const clickFn = jest.fn();
        const component = shallow(<InfoModal onClose={clickFn}>Error</InfoModal>);
        component.find("Button").simulate('click');

        expect(clickFn).toHaveBeenCalledTimes(1);
    });
});
