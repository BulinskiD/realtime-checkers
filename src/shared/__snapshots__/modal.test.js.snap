// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoModal should render correctly with given props 1`] = `
<Bootstrap(Modal)
  onHide={[MockFunction]}
>
  <ModalHeader
    closeButton={false}
    closeLabel="Close"
  >
    <ModalTitle>
      Test
    </ModalTitle>
  </ModalHeader>
  <ModalBody>
    <div
      className="text-center"
    >
      Error
    </div>
  </ModalBody>
  <ModalFooter>
    <Button
      active={false}
      disabled={false}
      onClick={[MockFunction]}
      type="button"
      variant="danger"
    >
      Button
    </Button>
  </ModalFooter>
</Bootstrap(Modal)>
`;
